skstore/ts/tests/tests.ts

```diff
@@ -0,0 +1,509 @@
+  check("testMapReduceInsert", output.select({}, ["id", "v"]), [
+    { id: 0, v: 2 },
+    { id: 1, v: 3 },
+  ]);
+  input.update([0, 1], { v: 2 });
+  check("testMapReduceUpdate", output.select({}, ["id", "v"]), [
+    { id: 0, v: 3 },
+    { id: 1, v: 3 },
+  ]);
+  input.deleteWhere({ id: 3 });
+  check("testMapReduceRemove", output.select({}, ["id", "v"]), [
+    { id: 0, v: 3 },
+    { id: 1, v: 1 },
+  ]);
+}
+
+//// testMultiMap1
+
+async function testMultiMap1Init(
+  skstore: SKStore,
+  input1: TableHandle<[number, string]>,
+  input2: TableHandle<[number, string]>,
+  output: TableHandle<[number, number, number]>,
+) {
+  const eager1 = input1.map((row, _occ) => {
+    return Array([row[0], parseInt(row[1])]);
+  });
+  const eager2 = input2.map((row, _occ) => {
+    return Array([row[0], parseInt(row[1])]);
+  });
+  const eager3 = skstore.multimap([
+    [
+      eager1,
+      (key, it) => {
+        const v = it.first();
+        return Array([[0, key], v]);
+      },
+    ],
+    [
+      eager2,
+      (key, it) => {
+        const v = it.first();
+        return Array([[1, key], v]);
+      },
+    ],
+  ]);
+  eager3.mapTo(output, (key, it) => {
+    return [key[0], key[1], it.first()];
+  });
+}
+
+async function testMultiMap1Run(
+  input1: Table<[number, number]>,
+  input2: Table<[number, number]>,
+  output: Table<[number, number, number]>,
+) {
+  input1.insert([[1, 10]], true);
+  input2.insert([[1, 20]], true);
+  check("testMultiMapInit", output.select({}, ["src", "id", "v"]), [
+    { src: 0, id: 1, v: 10 },
+    { src: 1, id: 1, v: 20 },
+  ]);
+  input1.insert([[2, 3]], true);
+  input2.insert([[2, 7]], true);
+  check("testMultiMapInsert", output.select({}, ["src", "id", "v"]), [
+    { src: 0, id: 1, v: 10 },
+    { src: 0, id: 2, v: 3 },
+    { src: 1, id: 1, v: 20 },
+    { src: 1, id: 2, v: 7 },
+  ]);
+  input1.deleteWhere({ id: 1 });
+  check("testMultiMapDelete", output.select({}, ["src", "id", "v"]), [
+    { src: 0, id: 2, v: 3 },
+    { src: 1, id: 1, v: 20 },
+    { src: 1, id: 2, v: 7 },
+  ]);
+}
+
+//// testMultiMapReduce
+
+function testMultiMapReduceInit(
+  skstore: SKStore,
+  input1: TableHandle<[number, string]>,
+  input2: TableHandle<[number, string]>,
+  output: TableHandle<[number, number]>,
+) {
+  const eager1 = input1.map((row, _occ) => {
+    return Array([row[0], parseInt(row[1])]);
+  });
+  const eager2 = input2.map((row, _occ) => {
+    return Array([row[0], parseInt(row[1])]);
+  });
+  const eager3 = skstore.multimapReduce(
+    [
+      [
+        eager1,
+        (key, it) => {
+          const v = it.first();
+          return Array([key, v]);
+        },
+      ],
+      [
+        eager2,
+        (key, it) => {
+          const v = it.first();
+          return Array([key, v]);
+        },
+      ],
+    ],
+    new Sum(),
+  );
+  eager3.mapTo(output, (key, it) => {
+    return [key, it.first()];
+  });
+}
+
+async function testMultiMapReduceRun(
```

Comment:
Why is this one async?

Reply:
It's to provide a global mechanism for all tests.
For async tests, It's used to wait for the async calls before doing the checks.
It's possible to split the tests in two categories.